crates/experimentation_platform/src/db/models.rs

```diff
@@ -25,6 +28,39 @@ pub enum ExperimentStatusType {
     INPROGRESS,
 }
 
+#[derive(Deserialize, Serialize, Clone, PartialEq, Debug)]
+pub enum VariantType {
+    CONTROL,
+    EXPERIMENTAL,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct Variant {
+    pub id: String,
+    pub variant_type: VariantType,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub context_id: Option<String>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub override_id: Option<String>,
+    pub overrides: Exp<Overrides>,
+}
+
+#[derive(
+    Debug, Clone, Serialize, Deserialize, AsExpression, FromSqlRow, JsonFromSql, JsonToSql,
+)]
+#[diesel(sql_type = Json)]
+pub struct Variants(Vec<Variant>);
+
+impl Variants {
+    pub fn new(data: Vec<Variant>) -> Self {
+        Self(data)
+    }
+
+    pub fn into_inner(self) -> Vec<Variant> {
+        self.0
+    }
+}
```

Comment:
You can use strum::macros Deref to handle into_inner and you may not need the new function

Reply:
using that adds more complexity as to add that you need to implement Copy trait, and then String does not implement Copy, so its not possible to add Deref for this

Reply:
Can you check default_config API file? It has this as a struct
```
#[derive(Debug, Deserialize, AsRef, Deref, DerefMut, Into)]
#[serde(try_from = "String")]
pub struct DefaultConfigKey(String);
```

You can probably do this

Reply:
Also for tenant struct
```
#[derive(Deref, DerefMut, Clone, Debug)]
pub struct Tenant(pub String);
```